li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -265,7 +265,9 @@ synchronized public void assign(Collection<TopicPartition> partitions) {
     _numClustersToConnectTo = clusterToTopicPartitionsMap.size();
     _nextClusterIndexToPoll = 0;
     for (Map.Entry<ClusterDescriptor, Set<TopicPartition>> entry : clusterToTopicPartitionsMap.entrySet()) {
-      getOrCreatePerClusterConsumer(consumers, entry.getKey()).assign(entry.getValue());
+      LiKafkaConsumer<K, V> curConsumer = createPerClusterConsumer(entry.getKey());
+      curConsumer.assign(entry.getValue());
```

Comment:
Can you add a TODO for re-subscribing topic partitions when subscription is supported?